### PR TITLE
[TIMOB-19769] (5_2_X) Android: Fix for fullscreen behavior in Camera

### PR DESCRIPTION
**Pull request by ashcoding**
Jira: https://jira.appcelerator.org/browse/TIMOB-19769
PR https://github.com/appcelerator/titanium_mobile/pull/7552 caused an edge case issue for Camera.

Code to test:-

``` javascript
var self = Ti.UI.createWindow();
    var cameraOverlay = Titanium.UI.createView({
        backgroundColor: '#FF00FFAA'
    });

    var cameraOverlayButton = Titanium.UI.createImageView({
        bottom: 100,
        image: '/appcelerator_small.png',
    });
    cameraOverlay.add(cameraOverlayButton);

    cameraOverlay.addEventListener('click', function(e){
        Ti.API.info("taking pic");
        Ti.Media.takePicture();
    });

    self.addEventListener("postlayout", postlayoutHandler);
    function postlayoutHandler(e) {
        self.removeEventListener("postlayout", postlayoutHandler);
        Titanium.Media.showCamera({
            overlay: cameraOverlay,
            showControls: false,
            success: function(e) {
                  Ti.API.info('# SUCCESS : '+ JSON.stringify(e));
            },
            cancel: function(e) {
                  Ti.API.info('# CANCEL : '+e.error);

            },
            error: function(e) {
                  Ti.API.info('# ERROR : '+ e.code +' : '+ e.error);

            },
            mediaTypes:Ti.Media.MEDIA_TYPE_PHOTO,
        });
    }
    self.open();
```

### CODE REVIEW

**Comment by ashcoding** [comment]
This is for 5_2_X. Will add a Master branch PR once this is approved.

File: android/modules/media/src/java/ti/modules/titanium/media/TiCameraActivity.java [hunk 1]
```diff
@@ -134,7 +137,7 @@ protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec)
 	@Override
 	public void onCreate(Bundle savedInstanceState)
 	{
-		setFullscreen(true);
+		setFullscreenForCamera(true);
```
Comment (sriks) [inline android/modules/media/src/java/ti/modules/titanium/media/TiCameraActivity.java]: Since base class should be agnostic of derived classes, adding camera specific method in base class is not advised. 

`getWindow().addFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN);` can be moved here. Also this is being used specifically for TiCameraActivity only.

File: android/modules/media/src/java/ti/modules/titanium/media/TiCameraActivity.java [hunk 2]
```diff
@@ -161,6 +164,38 @@ public void onCreate(Bundle savedInstanceState)
 
 	}
 
+	// This is overriden to prevent using the FullScreen method which is not supported for Camera use.
+	@Override
+	protected void windowCreated(Bundle savedInstanceState) {
```
Comment (sriks) [inline android/modules/media/src/java/ti/modules/titanium/media/TiCameraActivity.java]: This override may not be required. 

TiCameraActivity is a defined activity in manifest file which is used only for camera preview. So fullscreen is always `false` here
https://github.com/ashcoding/titanium_mobile/blob/TIMOB-19769-B/android/titanium/src/java/org/appcelerator/titanium/TiBaseActivity.java#L500

So override is unnecessary.

**Comment by sriks** [comment]
@ashcoding please address the review comments.

File: android/modules/media/src/java/ti/modules/titanium/media/TiCameraActivity.java (hunk 1 above)
Comment (ashcoding) [inline android/modules/media/src/java/ti/modules/titanium/media/TiCameraActivity.java]: I moved the `setFullscreenForCamera` method into this class to address this

**Comment by ashcoding** [comment]
Comments addressed @sriks. Please review again. Thank you.

File: android/modules/media/src/java/ti/modules/titanium/media/TiCameraActivity.java [hunk 3]
```diff
@@ -161,6 +164,13 @@ public void onCreate(Bundle savedInstanceState)
 
 	}
 
+	public void setFullscreenForCamera(boolean fullscreen)
```
Comment (sriks) [inline android/modules/media/src/java/ti/modules/titanium/media/TiCameraActivity.java]: checking for truth value is not required since we never set back from fullscreen.

**Comment by ashcoding** [comment]
@sriks, Comments addressed.

**Comment by sriks** [comment]
PR reviewed and merged!
